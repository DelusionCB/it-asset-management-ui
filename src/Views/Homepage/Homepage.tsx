import React from 'react';
import Directory from '../../Components/Directory/Directory';
import './index.scss'
import {useTranslation} from 'react-i18next';

function Homepage (): JSX.Element {
    const {t} = useTranslation()
    return (
        <div className='container home-container'>
            <div className='container-md'>
                <h1>{t('home.header')}</h1>
                <p>
                    Mappi-järjestelmä auttaa hallinnoimaan sopimuksia, sovelluksia, palvelinasioita ja arkistoa.
                </p>
            </div>
            <Directory />
        </div>
    );
}

export default Homepage;
