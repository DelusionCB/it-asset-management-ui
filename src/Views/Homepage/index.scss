.home-container {
    padding: 10px;
    h1 {
        font-size: 2rem;
        word-break: break-all;
    }
}
