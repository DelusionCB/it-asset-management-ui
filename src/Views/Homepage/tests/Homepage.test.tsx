import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import Homepage from '../Homepage';
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'
import DirectoryProvider from '../../../Context/DirectoryContext';

describe('<Homepage />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <DirectoryProvider>
                    <Homepage />
                </DirectoryProvider>
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
