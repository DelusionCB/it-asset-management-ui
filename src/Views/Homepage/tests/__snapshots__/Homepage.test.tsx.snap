// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage /> Snapshot renders correctly 1`] = `
<div
  className="container home-container"
>
  <div
    className="container-md"
  >
    <h1>
      home.header
    </h1>
    <p>
      Mappi-järjestelmä auttaa hallinnoimaan sopimuksia, sovelluksia, palvelinasioita ja arkistoa.
    </p>
  </div>
  <div
    animation="border"
    className="spinner-border"
    role="status"
  >
    <span
      className="visually-hidden"
    >
      Ladataan...
    </span>
  </div>
</div>
`;
